src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java

```diff
@@ -128,18 +138,38 @@ public String get() {
 
         icon = find("icon", ItemIcon.class);
         if (icon != null) {
+            icon.bindIcon(new ReadOnlyBinding<TextureRegion>() {
+                @Override
+                public TextureRegion get() {
+                    if (cameraTargetSystem.isTargetAvailable()) {
+                        if (!cameraTargetSystem.isBlock()) {
+                            EntityRef targetEntity = cameraTargetSystem.getTarget();
+                            GetTooltipIconEvent getTooltipIconEvent = new GetTooltipIconEvent();
+                            targetEntity.send(getTooltipIconEvent);
```

Comment:
This is a much better pattern than the hardcoding I did for block entities.  I suppose the natural evolution would use a ```GetTooltipMeshEvent``` to get the mesh for blocks (not now,  just thoughts for the future)